amd/src/mod_hsuforum_loader.js

```diff
@@ -42,28 +43,59 @@ define(['jquery'], function ($) {
             posts.push(footerReply);
         }
 
+        var cookie = "Reply=yes";
         if (posts) {
             $(posts).each(function(){
                 const postObserver = new MutationObserver(() => {
                     let form = $(this).find('form');
-
                     if (form) {
                         let formTextarea = $(form).find('.hsuforum-textarea');
                         $(form).on('submit', () => {
                             // Check for form errors
                             if ($(formTextarea).text() != 0) {
                                 document.body.dispatchEvent(spinnerStartEvent);
+                                var postid = $('input:hidden[name=reply]').val(); //forms current hidden reply value, replyto
+                                var markasread = ajax.call([{
+                                    headers: "max-age=1000",
+                                    methodname: 'mod_hsuforum_mark_single_post_read',
+                                    args: {postid: postid},
+                                }]);
+                                markasread[0].done(function (response) {
+                                    location.reload();
+                                    document.cookie = cookie;
+                                        setTimeout(function () {
```

Comment:
Why the timeouts?

Reply:
used the time outs to get the updated url which is the new post id to mark that new post also as read